tests for join class implemented; changed implementation in join use …

yeddddy opened the pull request:
…case interactor

theflickerman404 reviewed (changes requested):
- You should avoid checked exceptions in general, but especially in the entity classes, if there's an invalid argument passed, its not the job of the entity to throw a proper error, instead of catching the exception, just let it crash the program and let the traceback tell us where we went wrong
- Error handling should be done at the interactor level and conveyed by calling the appropriate presenter
- Interactor looks good besides the try catch block, see above comments on how to resolve this, complies with clean architecture, and is diadatic, well done
- Tests look good

yeddddy commented:
Good point! Exceptions should not be handled on the entity level in general, but I think I simply forgot the IllegalArgumentException is a child class of RunTimeException lol.
Fixed the method's signature in Account and modified the interaction accordingly (removed the try-catch block and changed to if-checks).
Hope it looks good now!

theflickerman404 reviewed (approved):
Perfect, thank you
